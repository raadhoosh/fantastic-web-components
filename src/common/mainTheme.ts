import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#4cae4c',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
        default: '#6c757d'
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    unit: {
        padding: .1,
        font: .1,
        margin: .1,
    },
    button: {
        margin: '5px 3px',
        padding: '.375rem .75rem',
        radius: '5%',
        fontsize: '1rem',
        color: '#fff',
        border: '5px',
        shadow: '0 0 0 0.2rem #color',
        transition: `color .15s ease-in-out,
        background-color.15s ease-in-out,
        border-color .15s ease-in-out,
        box-shadow .25s ease-in-out`
    },
    alert: {
        margin: '5px 0px',
        padding: '.75rem 1.25rem',
        fontsize: '12',
        radius: '5px',
        border: '1px solid transparent',
        color: '#fff',
    },
    input: {
        fontsize: '12',
        border: '1px solid',
    },
    textarea: {
        fontsize: '12',
        width: 'auto',
        height: '50px',
    },
    label: {
        fontsize: '12',
        borderStyle: 'solid',
        borderSize: 0,
        color: '#fff',
    },
    badge: {
        fontsize: '.756rem',
        borderStyle: 'solid',
        borderSize: '1px',
        color: '#fff',
        width: 'auto',
        height: 'auto',
        radius: '10px',
    },
    h1: {
        fontsize: '2em',
        color: '#000'
    },
    h2: {
        fontsize: '1.5em',
        color: '#000'
    },
    h3: {
        fontsize: '1.17em',
        color: '#000'
    },
    h4: {
        fontsize: '1.33em',
        color: '#000'
    },
    h5: {
        fontsize: '0.83em',
        color: '#000'
    },
    h6: {
        fontsize: '0.67em',
        color: '#000'
    },
    card: {
      width: '499px'
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
