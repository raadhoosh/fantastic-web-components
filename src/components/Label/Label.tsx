import styled from '../../common/styled';
import * as React from 'react';
/**
 *  properties
 */
interface IProps {
  /** This is a primary label it out!. */
  primary?: boolean;
  /** This is a secondary label it out!. */
  secondary?: boolean;
  /** This is a success label it out!. */
  success?: boolean;
  /** This is a info label it out!. */
  info?: boolean;
  /** This is a warning label it out!. */
  warning?: boolean;
  /** This is a danger label it out!. */
  danger?: boolean;
  /** This is a light label it out!. */
  light?: boolean;
  /** This is a dark label it out!. */
  dark?: boolean;
  /** This is a text color label it out! */
  children?: any;
  /** Description of prop "primary". */
  backgroundColor?: string;
  color?: string;
  margin?: string;
  theme?: any;
}

const EmLabel = styled('label')(
  (props: (IProps)) => {
    const base_label = {
      display: 'inline',
      padding: '.2em .6em .3em',
      fontSize: '75%',
      fontWeight: 700,
      lineHeight: 1,
      verticalAlign: 'baseline',
      borderRadius: '.25em',
      margin: props.margin ? props.margin : 2,
      color: props.color ? props.color : '#fff',
      backgroundColor: 'transparent'
    };
    if (props.primary) {
      base_label.backgroundColor = props.theme.color.primary;
      return base_label;
    }
    if (props.secondary) {
      base_label.backgroundColor = props.theme.color.secondary;
      return base_label;
    }
    if (props.success) {
      base_label.backgroundColor = props.theme.color.success;
      return base_label;
    }
    if (props.info) {
      base_label.backgroundColor = props.theme.color.info;
      return base_label;
    }
    if (props.danger) {
      base_label.backgroundColor = props.theme.color.danger;
      return base_label;
    }
    if (props.warning) {
      base_label.backgroundColor = props.theme.color.warning;
      return base_label;
    }
    if (props.light) {
      base_label.backgroundColor = props.theme.color.light;
      base_label.color = '#000';
      return base_label;
    }
    if (props.dark) {
      base_label.backgroundColor = props.theme.color.dark;
      return base_label;
    }

    base_label.backgroundColor = props.backgroundColor ? props.backgroundColor : 'transparent';
    return base_label;
  }
);

/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const Label = (props: IProps) => {

  return (
    <EmLabel {...props}> {props.children} </EmLabel>
  );
};

export default Label;
