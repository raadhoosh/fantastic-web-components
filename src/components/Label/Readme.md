Label example:

```js
<Label primary >primary</Label>
<Label secondary >secondary</Label>
<Label success >success</Label>
<Label info >info</Label>
<Label warning >warning</Label>
<Label danger >danger</Label>
<Label light >light</Label>
<Label dark >dark</Label>
```

Label example Badge:

```js
<Label primary badge >23</Label>
<Label secondary badge>12</Label>
<Label success badge>2</Label>
<Label info badge>66</Label>
<Label warning badge>4</Label>
<Label danger badge>33</Label>
<Label light badge >89</Label>
<Label dark  badge>1</Label>
```