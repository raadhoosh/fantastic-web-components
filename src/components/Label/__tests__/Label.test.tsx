import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Label from '../Label';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const lbl = renderer.create(<Label theme={mainTheme} success >ok</Label>).toJSON();
    expect(lbl).toMatchSnapshot();
});
