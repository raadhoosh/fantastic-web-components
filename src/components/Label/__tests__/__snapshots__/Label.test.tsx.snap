// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<label
  className="css-e1ymif"
>
   
  ok
   
</label>
`;
